src/main/java/net/tropicraft/core/client/ScubaHandler.java

```diff
@@ -31,13 +35,39 @@ public ScubaHandler() {
 		flipperSpeedMap.put(ItemRegistry.pinkFlippers, 1f);
 		flipperSpeedMap.put(ItemRegistry.yellowFlippers, 1.5f);
 	}
+	
+	@SubscribeEvent
+	public void onRenderViewTick(EntityViewRenderEvent.CameraSetup event) {
+		if(event.getEntity().equals(Minecraft.getMinecraft().player)) {
+			if(Minecraft.getMinecraft().player != null && !Minecraft.getMinecraft().player.isDead) {
+					EntityPlayer p = Minecraft.getMinecraft().player;
+					PlayerSwimData d = getData(p);
+					//if(p.isInWater())
+					event.setRoll(-d.currentRotationRoll*0.25f);
+					//event.setPitch(event.getPitch()+(d.currentHeadPitchOffset*0.25f));
+				if (Minecraft.getMinecraft().gameSettings.thirdPersonView == 0) {
+				//	updateSwimDataAngles(p);
+				}
+			}
+		}
+	}
+	
+	@SubscribeEvent
+	public void onRenderViewTick(RenderHandEvent event) {
+		
+		
+	}
+	
+	
 
 	@SubscribeEvent
 	public void onRenderTick(RenderWorldLastEvent event) {
 		if(Minecraft.getMinecraft().player != null && !Minecraft.getMinecraft().player.isDead) {
-			if (Minecraft.getMinecraft().gameSettings.thirdPersonView == 0) {
+			if (Minecraft.getMinecraft().gameSettings.thirdPersonView == 0 || Minecraft.getMinecraft().currentScreen != null) {
 				EntityPlayer p = Minecraft.getMinecraft().player;
+				if(!p.isInWater())
```

Comment:
This is sketchy. Should this encompass both of these? just one?

Reply:
just the one, it will be more obvious in the future when i intend to do this